enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,823 @@
+```
+type: IPsAllocated
+status: "False"
+reason: IPAddressConflict
+message: "Requested IP 192.168.1.5 is already assigned in the network"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+```
+type: IPsAllocated
+status: "False"
+reason: RequestedIPOutsideSubnet
+message: "Requested IP 200.168.1.5 is outside the UDN subnet 192.168.0.0/16"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+```
+type: IPsAllocated
+status: "False"
+reason: ReservedIPRequested
+message: "The user requested IP address 192.168.200.123 which was explicitly excluded by the admin"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The management cluster should have no reason to import an existing VM from
+another virtualization platform into OpenShift Virtualization.
+
+It makes more sense to import VMs into the hosted clusters, in order to provide
+tenants with routed ingress into their hosted cluster workloads.
+
+There should be no hypershift platform-specific considerations with this
+feature.
+
+#### Standalone Clusters
+
+Full support.
+
+#### Single-node Deployments or MicroShift
+
+Full support; there might be some increased resource usage when scaling up
+number of VMs requesting IP and MAC address, since it will mean more processing
+in OVN-Kubernetes, and more CRs to reconcile.
+
+### Implementation Details/Notes/Constraints
+
+TODO
+
+### Risks and Mitigations
+
+TODO
+
+### Drawbacks
+
+The biggest drawback we have to implementing this feature is lack of clear
+asks from customers - we're pretty much guessing what they would want to use.
+
+We do not know (for instance):
+- is this feature only for VMs ?
+- is this feature only about importing VMs ? Should we allow creating new VMs
+  with dedicated MAC / IP / gateway requests ?
+- are the end users (admins) after a centralized IP management alternative ?
+  Do they want to have a single place to check their network address
+  assignments ?
+
+## Alternatives (Not Implemented)
+
+### Requesting IP addresses on a VM's interfaces
+An option which would provide a simpler implementation (and better / more
+explicit UX) would be to modify the VMI specification, adding an attribute for
+the user (or the tool templating the VM - e.g. MTV) to specify which IP
+addresses they want the VM's interface to take.
+
+It would be updating the
+[v1.Interface](https://kubevirt.io/api-reference/main/definitions.html#_v1_interface)
+with an ipAddresses array. This attribute is defined in
+`VMI.Spec.Domain.Devices.Interfaces`.
+
+This option (despite being the simplest to implement) is not taken into
+consideration since we're already past the KubeVirt 1.5 API freeze - as a
+result, we wouldn't be able to get this in OpenShift 4.20. This should be taken
+into account for a future release though.
+
+### Defining the MAC to IPs association
+There are two alternatives to persist this information: a
```

Comment:
hmm wait I am not sure I understand this part. Is this _only for MTV_ section?
i.e today MTV does the "annotation on the VM spec" for IP v/s MAC on template approach. Is this section about how MTV will store the information for all IPs/MACs of all VMs? If so why does it need to separately persist this info? I thought MTV had access to the disk where the VM's networking info can be fetched right?
I think I might be missing the information on what is the purpose to have MTV store this association information

Reply:
Resolving this, since I didn't receive any request for deleting this info.